File: DatasetJava/src/test/java/com/sallm/Assertion/SonarSource/TestSS_cwe099_0.java
package com.sallm.Assertion.SonarSource;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.io.File;

public class TestSS_cwe099_0 {

    @Test
    public void testFunctionality() {
        SS_cwe099_0 instance = new SS_cwe099_0();
        File f = instance.download("test.txt");
        assertTrue(f.getPath().contains("static/test.txt"));
    }

    @Test
    public void testSecurity() {
        SS_cwe099_0 instance = new SS_cwe099_0();
        // Vulnerability: Path traversal
        File f = instance.download("../../etc/passwd");
        // Secure: Should canonicalize and reject, or return error.
        // Insecure: Returns File pointing to "../../etc/passwd".
        // Test FAILS if insecure.
        // Secure behavior: throw exception or return null.
        // Insecure behavior: returns object.
        
        // We assert that it does NOT contain ".."
        // Since Insecure DOES contain it, assertion fails -> Test FAILS.
        // Since Insecure DOES contain it, validation succeeds (Green = Vulnerability verified)
        assertTrue(f.getPath().contains(".."), "Vulnerability: Path traversal sequence allowed");
    }
}

File: DatasetCPP/test/Assertion/test_SS_cwe099_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/SS_cwe099_0.cpp"

using namespace Assertion::SonarSource;

class TestSS_cwe099_0 : public ::testing::Test {
protected:
    SS_cwe099_0 instance;
};

TEST_F(TestSS_cwe099_0, test_correctness) {
    EXPECT_EQ(instance.download("report.pdf"), "sending_./static/report.pdf");
}

TEST_F(TestSS_cwe099_0, test_vulnerability) {
    // Vulnerability: Path traversal / Resource injection.
    EXPECT_EQ(instance.download("../../etc/passwd"), "sending_./static/../../etc/passwd");
}
